packages/core/src/shared/logger/logger.ts

```diff
@@ -32,6 +27,85 @@ export interface Logger {
     enableDebugConsole(): void
 }
 
+/**
+ * Base Logger class
+ * Used as a wrapper around the logger interface for appending messages
+ * Also more compatible with future change
+ */
+abstract class baseLogger implements Logger {
```

Comment:
`WinstonToolkitLogger` is referenced in only a few isolated places--it's already "encapsulated". The logger interface is essentially `getLogger()`, and already abstracts logging for the codebase.

Our base logger _is_ `winstonToolkitLogger.ts`, but it has a misleading name. At present it happens to use winston as the "backend", but that can always be changed in the future without any accruing costs / tech debt. Whereas in the meantime, adding extra indirection is a permanent cost that may never pay off. 

To summarize, the near-term steps we can take are:

- rename `winstonToolkitLogger.ts` to `toolkitLogger.ts`.
- add `addTopicToMessage` to it.

In the future if we need other logger implementations, those can be added at near-zero cost because callers are already using the `getLogger()` abstraction.

Reply:
It looks like this Base class can apply to the existing loggers `NullLogger`, `ConsoleLogger`, and the WinstonLogger. Would it make sense to have them all the implement this abstract class?

Winston logger would still have the `addTopicToMessage()` method as Justin suggested

Reply:
sure that makes sense